Задание 1-3

Divium-l opened the pull request:
1. Написана программа
2. Создана блок-схема алгоритма

Divium-l committed:
Задание 1-3

1. Написана программа
2. Создана блок-схема алгоритма

MarinaPaley reviewed (changes requested):
Я в прошлый раз не подумала, что Вы всегда теперь будете применять разработанный метод input(). У него больше одной ответственности: считывает и парсит. Разбейте метод на два. Также не понятна мне Ваша любовь к отрицательным вводам. Запретите явно, когда парсите, вводить значение меньше нуля.

Divium-l committed:
Задание 1-3 fix-1

Функция input() разбита на две функции

MarinaPaley commented on task-1-3/task-1-3.cpp:
```diff
@@ -0,0 +1,80 @@
+
+	force = calcGravityForce(mass);
+
+	cout << endl << endl;
```
Исправьте на `\n'
Добавьте для всех функций документацию

MarinaPaley commented on task-1-3/task-1-3.cpp:
```diff
@@ -0,0 +1,80 @@
+	bool isValid, isFloat;
+	int i = 0;
+
+	isValid = true;
```
Объявляйте и сразу инициализируйте переменные там, где возможно

MarinaPaley commented on task-1-3/task-1-3.cpp:
```diff
@@ -0,0 +1,80 @@
+	return stod(str);
+}
+
+bool isValidDouble(string str, bool allowNegative)
```
`const bool...`

MarinaPaley commented on task-1-3/task-1-3.cpp:
```diff
@@ -0,0 +1,80 @@
+
+}
+
+double input(bool allowNegative) {
```
`const bool ...'

MarinaPaley commented on task-1-3/task-1-3.cpp:
```diff
@@ -0,0 +1,80 @@
+
+double calcGravityForce(const double mass)
+{
+	return 9.81 * mass;
```
Введите константу `g = 9.81`

MarinaPaley reviewed (changes requested):
Небольшие замечания

Divium-l committed:
Задание 1-3 experimental

1. Функции проверки ввода вынесены в отдельную библиотеку  (если так можно сказать). Функции были изменены, чтобы  их можно было использовать в других заданиях без изменений.
Также была добавлена функция для ввода целочисленных значений.
2. Небольшие исправления

MarinaPaley commented on task-1-3/input_lib.cpp:
```diff
@@ -0,0 +1,108 @@
+{
+	public:
+
+		static double inputDouble(const bool allowNegative = true, const string valueInputMessage = "Value: ", const string valueIncorrectMessage = "Incorrect value!Please try again\n")
```
`...const string& mesage = "Value; ", const string& errorMessage = "..."`

MarinaPaley commented on task-1-3/input_lib.cpp:
```diff
@@ -0,0 +1,108 @@
+			return stod(str);
+		}
+	
+		static int inputInt(const bool allowNegative = true, const string valueInputMessage = "Value: ", const string valueIncorrectMessage = "Incorrect value!Please try again\n")
```
`string` - объект, лучше передавать не копируя - по ссылке

MarinaPaley reviewed (changes requested):
Добавьте документацию
объекты передавайте по ссылке
может понравится для [int](https://github.com/MarinaPaley/Demo-CPP/blob/feature/25/Task%204-1/main.cpp#L365)

Divium-l committed:
Задание 1-3 fix-2

1. Добавлена документация
2. Объекты string передаёются по ссылке

MarinaPaley commented on task-1-3/input_lib.cpp:
```diff
@@ -0,0 +1,134 @@
+
+using namespace std;
+
+class input
```
Давайте всё же если это _процедурное программирование_, то это будет `namespace`, а не `class` &ndash; Вам всё равно, а мне приятно.

Просто статические методы класса &ndash; это скорее костыль ООП, но у нас-то тут не обязательно ООП, а как раз наоборот &ndash; процедурное программирование. Здесь-то как раз _просто так_ висящие методы _разрешены_.

Хотя Вы там уже́ начали играться с доступом.
